activityLogsApp/UpdateNSGFlows.cs

```diff
@@ -213,63 +219,75 @@ static async Task enable_flow_logs(VNETApiResult vnetresult, Dictionary<string,
         	try
             {
+				dynamic myObject = new JObject();
+            	HttpRequestMessage req = new HttpRequestMessage(HttpMethod.Get, String.Format(query_flow_logs_url, subs_id,resourceGroupName, nw_name));
+                req.Headers.Accept.Clear();
+                req.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+                HttpResponseMessage response = await SingleHttpClientInstance.sendApiRequest(req, token);
 
-            	dynamic myObject = new JObject();
-            	myObject.targetResourceId = vnet.id;
-            	HttpRequestMessage checkReq = new HttpRequestMessage(HttpMethod.Post, String.Format(query_flow_logs_url, loc_nw));
-            	var content = new StringContent(myObject.ToString(), Encoding.UTF8, "application/json");
-            	checkReq.Content = content;
-                checkReq.Headers.Accept.Clear();
-                checkReq.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
-
-                HttpResponseMessage check_response = await SingleHttpClientInstance.sendApiPostRequest(checkReq, token);
-
-                if (check_response.IsSuccessStatusCode)
+                log.LogInformation("Entered into the check and enable flow request function inside try");
+                if (response.IsSuccessStatusCode)
 				{
-				    string check_data =  await check_response.Content.ReadAsStringAsync();
-				    var check_result = JsonConvert.DeserializeObject<FlowLogStatusResponse>(check_data);
-				    if(check_result.properties.enabled){
-				    	return "false";
+				    string check_data =  await response.Content.ReadAsStringAsync();
+				    var json = JObject.Parse(check_data);
+				    HashSet<string> enabledVnetIds = new HashSet<string>();
+				    foreach (var flowLog in json["value"]){
+				        var targetResourceId = flowLog["properties"]["targetResourceId"].ToString();
+                        if(targetResourceId.Contains("virtualNetworks")){
+                            enabledVnetIds.Add(targetResourceId);
+                        }
+				    }
+				    if(!enabledVnetIds.Contains(vnet.id)){
+				        dynamic properties = new JObject();
+                        properties.storageId = storageId;
+                        properties.targetResourceId = vnet.id;
+                        properties.enabled = true;
+                        dynamic retention = new JObject();
+                        retention.days = 1;
+                        retention.enabled = true;
+                        properties.retentionPolicy = retention;
+                        myObject.properties = properties;
+                        var content = new StringContent(myObject.ToString(), Encoding.UTF8, "application/json");
+
+                        log.LogInformation( "Entered into the check and enable flow request function and enabling the flow logs for this vnet");
+                        string flowLogName = vnet.name + "-vnet-flowlogs";
```

Comment:
flowLogName should have resourceGroup name in it

Reply:
remove -vnet text

Reply:
Updated